File: libraries/OpenCL/src/main/scala/com/nativelibs4java/scalacl/NIOUtils.scala
package com.nativelibs4java.scalacl
import java.nio._

object NIOUtils {
  implicit def array2Buffer(a: Array[Int   ]) = IntBuffer.wrap(a)
  implicit def array2Buffer(a: Array[Short ]) = ShortBuffer.wrap(a)
  implicit def array2Buffer(a: Array[Long  ]) = LongBuffer.wrap(a)
  implicit def array2Buffer(a: Array[Byte  ]) = ByteBuffer.wrap(a)
  implicit def array2Buffer(a: Array[Char  ]) = CharBuffer.wrap(a)
  implicit def array2Buffer(a: Array[Double]) = DoubleBuffer.wrap(a)
  implicit def array2Buffer(a: Array[Float ]) = FloatBuffer.wrap(a)

  def directInts(n: Int) = ByteBuffer.allocateDirect(n * 4).asIntBuffer()
  def directShorts(n: Int) = ByteBuffer.allocateDirect(n * 2).asShortBuffer()
  def directLongs(n: Int) = ByteBuffer.allocateDirect(n * 8).asLongBuffer()
  def directBytes(n: Int) = ByteBuffer.allocateDirect(n)
  def directChars(n: Int) = ByteBuffer.allocateDirect(n * 2).asCharBuffer()
  def directDoubles(n: Int) = ByteBuffer.allocateDirect(n * 8).asDoubleBuffer()
  def directFloats(n: Int) = ByteBuffer.allocateDirect(n * 4).asFloatBuffer()

}

File: libraries/OpenCL/src/main/java/com/nativelibs4java/nio/NIOUtils.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package com.nativelibs4java.nio;
import java.nio.*;
/**
 *
 * @author ochafik
 */
public class NIOUtils {
	public static IntBuffer directInts(int size) {
        return ByteBuffer.allocateDirect(size * 4).order(ByteOrder.nativeOrder()).asIntBuffer();
    }

    public static LongBuffer directLongs(int size) {
        return ByteBuffer.allocateDirect(size * 8).order(ByteOrder.nativeOrder()).asLongBuffer();
    }

    public static ShortBuffer directShorts(int size) {
        return ByteBuffer.allocateDirect(size * 2).order(ByteOrder.nativeOrder()).asShortBuffer();
    }

    public static ByteBuffer directBytes(int size) {
        return ByteBuffer.allocateDirect(size).order(ByteOrder.nativeOrder());
    }

    public static FloatBuffer directFloats(int size) {
        return ByteBuffer.allocateDirect(size * 4).order(ByteOrder.nativeOrder()).asFloatBuffer();
    }

    public static DoubleBuffer directDoubles(int size) {
        return ByteBuffer.allocateDirect(size * 8).order(ByteOrder.nativeOrder()).asDoubleBuffer();
    }
}
